tsdb/engine/tsm1/compact.go

```diff
@@ -1660,15 +1664,25 @@ type tsmBatchKeyIterator struct {
 	// without decode
 	merged    blocks
 	interrupt chan struct{}
+
+	// maxErrors is the maximum number of errors to store before discarding.
+	maxErrors int
 }
 
-func (t *tsmBatchKeyIterator) AppendError(err error) {
-	t.errs = append(t.errs, err)
+func (t *tsmBatchKeyIterator) AppendError(err error) bool {
+	if t.maxErrors > len(t.errs) {
+		t.errs = append(t.errs, err)
+		// Was the error stored?
+		return true
+	} else {
+		// Was the error dropped
+		return false
+	}
```

Comment:
Should we keep a counter of the number of dropped errors? Then in `tsmBatchKeyIterator.Err` we can add a final error that says `x additional errors occurred`?

I'm envisioning people in the future saying, "I don't understand why there's so little data. This giant TSM file isn't _that_ corrupted, it only had 100 errors during compacting." Adding that final count of dropped errors would prevent that from happening.

Reply:
That's a really good idea! Working on it.